File: cpp/2021/prelim/p09.cpp
#include <bits/stdc++.h>

using namespace std;

int gcd(const int x, const int y) {
    return x == 0 ? y : gcd(y % x, x);
}

int lcm(const int x, const int y) {
    return x / gcd(x, y) * y;
}

int main() {
    int x, y, z;
    cin >> x >> y >> z;
    cout << lcm(x, lcm(y, z));
}

File: java/2021/prelim/P09.java
import java.util.Scanner;

public class P09 {

    public static int gcd(int x, int y) {
        while (x != 0) {
            int t = x;
            x = y % x;
            y = t;
        }
        return y;
    }

    public static int lcm(int x, int y) {
        return x / gcd(x, y) * y;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int x = scanner.nextInt();
        int y = scanner.nextInt();
        int z = scanner.nextInt();
        System.out.println(lcm(x, lcm(y, z)));
    }
}
